update sample url for prep of rename of repo url (#6289)
sunpy/sunpy: changelog/6289.trivial.rst

> The sample-data URL will be changing from ``https://github.com/sunpy/sample-data/raw/master/sunpy/v1/`` to ``https://github.com/sunpy/data/raw/main/sunpy/v1/``.
> GitHub will keep the redirect alive for sometime but will eventually expire it.

Reviewer: ```suggestion
The primary sample-data URL will be changing from ``https://github.com/sunpy/sample-data/raw/master/sunpy/v1/`` to ``https://github.com/sunpy/data/raw/main/sunpy/v1/``.
GitHub will keep the redirect alive for sometime but will eventually expire it.
The ``data.sunpy.org`` mirror will continue to be available.
```

The primary sample-data URL will be changing from ``https://github.com/sunpy/sample-data/raw/master/sunpy/v1/`` to ``https://github.com/sunpy/data/raw/main/sunpy/v1/``.

> GitHub will keep the redirect alive for sometime but will eventually expire it.

Reviewer: ```suggestion
We expect GitHub to redirect from the old URL for sometime but will eventually expire it.
```
